rskj-core/src/test/java/co/rsk/mine/MinerUtilsTest.java

```diff
@@ -288,4 +293,30 @@ public void getAllTransactionsCheckOrderTest() {
         Assert.assertEquals(res.get(7).getGasPrice(), Coin.valueOf(100));
         Assert.assertEquals(res.get(8).getGasPrice(), Coin.valueOf(1));
     }
+
+    @Test
+    public void filterTransactions_getNonceThrowsNodeReferenceException_throwsRuntimeException() {
+        // Given
+        RskAddress senderAddressMock = new RskAddress(new byte[20]);
+        Transaction txMock = Tx.create(config, 0, 50000, 5, 0, 0, 0);
+        Mockito.when(txMock.getSender()).thenReturn(senderAddressMock);
+
+        List<Transaction> txList = new LinkedList<>();
+        txList.add(txMock);
+
+        Repository repositoryMock = Mockito.mock(Repository.class);
+        Mockito.doThrow(new NodeReferenceException("")).when(repositoryMock).getNonce(senderAddressMock);
+
+        // Then
+        try {
+            new MinerUtils().filterTransactions(new LinkedList<>(), txList, new HashMap<>(), repositoryMock, Coin.valueOf(1L));
+            fail("RuntimeException should be thrown!");
+        } catch (RuntimeException ex) {
```

Comment:
maybe inside the catch we could get its cause and assert it's a `NodeReferenceException`?
just to make sure that the cause is what we want (and if someone modifies something in the future, causing another `RuntimeException`, this test will fail)

Reply:
At this level MinerUtils throws a RuntimeException, the one that throws NodeReferenceException is NodeReference class. Maybe we can modify MinerUtils and instead of throwing another exception, just let it bubble up. I think It could be better.

Reply:
I agree! because in the end, our custom exception already is a `RuntimeException`

Reply:
At the end this changed a lot. Now we don't need this part.